manifests/server.pp

```diff
@@ -100,10 +101,20 @@
   }
 
   if $create_admin and ($service_ensure == 'running' or $service_ensure == true) {
-    mongodb::db { 'admin':
-      user     => $admin_username,
-      password => $admin_password,
-      roles    => $admin_roles,
+    if $admin_password {
+      mongodb::db { 'admin':
+        user     => $admin_username,
+        password => $admin_password,
+        roles    => $admin_roles,
+      }
+    } elsif $admin_password_hash {
+      mongodb::db { 'admin':
+        user          => $admin_username,
+        password_hash => $admin_password_hash,
+        roles         => $admin_roles,
+      }
+    } else {
+      fail('You did not specify a password or password hash for admin username')
     }
```

Comment:
Previously it didn't require a password and just let db.pp deal with it:
https://github.com/voxpupuli/puppet-mongodb/blob/a8a58578a52bc36ee771aadf021b23e00b2eeb63/manifests/db.pp#L29-L35

I'd prefer it passed both parameters and required the caller and `mongodb::db` to deal with it.

Reply:
I've update also the code to send password_hash along password to the ::db